runtime/runtime-core/native/src/aws/smithy/kotlin/runtime/time/InstantNative.kt

```diff
@@ -5,92 +5,118 @@
 
     public actual companion object {
         /**
          * Parse an ISO-8601 formatted string into an [Instant]
          */
         public actual fun fromIso8601(ts: String): Instant {
-            TODO("Not yet implemented")
+            var parsed = DateTimeFormats.ISO_8601.parse(ts)
+
+            // Handle leap seconds (23:59:59)
+            parsed.second = parsed.second?.coerceAtMost(59)
+
+            // Handle leap hours (24:00:00)
+            return if (parsed.hour == 24) {
+                parsed.hour = 0
+                Instant(parsed.toInstantUsingOffset() + 1.days)
+            } else {
+                Instant(parsed.toInstantUsingOffset())
+            }
         }
 
         /**
          * Parse an RFC5322/RFC-822 formatted string into an [Instant]
          */
-        public actual fun fromRfc5322(ts: String): Instant {
-            TODO("Not yet implemented")
-        }
+        public actual fun fromRfc5322(ts: String): Instant = Instant(KtInstant.parse(ts, DateTimeFormats.RFC_5322))
 
         /**
          * Create an [Instant] from its parts
          */
-        public actual fun fromEpochSeconds(seconds: Long, ns: Int): Instant {
-            TODO("Not yet implemented")
-        }
+        public actual fun fromEpochSeconds(seconds: Long, ns: Int): Instant = Instant(KtInstant.fromEpochSeconds(seconds, ns))
 
         /**
          * Parse a string formatted as epoch-seconds into an [Instant]
          */
-        public actual fun fromEpochSeconds(ts: String): Instant {
-            TODO("Not yet implemented")
-        }
+        @OptIn(FormatStringsInDatetimeFormats::class)
+        public actual fun fromEpochSeconds(ts: String): Instant = fromEpochSeconds(ts.toLong(), 0)
 
         /**
          * Create an [Instant] from the current system time
          */
-        public actual fun now(): Instant {
-            TODO("Not yet implemented")
-        }
+        public actual fun now(): Instant = Instant(Clock.System.now())
 
         /**
          * Create an [Instant] with the minimum possible value
          */
-        public actual val MIN_VALUE: Instant
-            get() = TODO("Not yet implemented")
+        public actual val MIN_VALUE: Instant = Instant(KtInstant.DISTANT_PAST)
 
         /**
          * Create an [Instant] with the maximum possible value
          */
-        public actual val MAX_VALUE: Instant
-            get() = TODO("Not yet implemented")
+        public actual val MAX_VALUE: Instant = Instant(KtInstant.DISTANT_FUTURE)
+    }
+
+    public override fun equals(other: Any?): Boolean = other is Instant && delegate == other.delegate
+    public override fun toString(): String = delegate.toString()
+    override fun hashCode(): Int {
+        var result = delegate.hashCode()
+        result = 31 * result + epochSeconds.hashCode()
+        result = 31 * result + nanosecondsOfSecond
+        return result
     }
```

Comment:
**Nit**: `epochSeconds` and `nanosecondsOfSecond` are derived from `delegate`. They shouldn't be part of the hash code calculation.